service/src/main/kotlin/fi/espoo/evaka/vasu/VasuController.kt

```diff
@@ -58,13 +58,14 @@ class VasuController(
                 throw Conflict("Cannot open a new vasu document while another is still active")
             }
 
-            tx.getVasuTemplate(body.templateId)?.let { template ->
-                if (!template.valid.includes(HelsinkiDateTime.now().toLocalDate())) {
-                    throw BadRequest("Template is not currently valid")
-                }
-            } ?: throw NotFound("template ${body.templateId} not found")
+            val template = tx.getVasuTemplate(body.templateId)
+                ?: throw NotFound("template ${body.templateId} not found")
 
-            tx.insertVasuDocument(childId, body.templateId)
+            if (!template.valid.includes(HelsinkiDateTime.now().toLocalDate())) {
+                throw BadRequest("Template is not currently valid")
+            }
+
+            tx.insertVasuDocument(childId, template)
```

Comment:
Nitpicking and just a personal preference but how about
```kt
tx.getVasuTemplate(body.templateId)
    ?.also { if (!it.valid.includes(HelsinkiDateTime.now().toLocalDate())) throw BadRequest("Template is not currently valid") }
    ?.let { tx.insertVasuDocument(childId, it) }
    ?: throw NotFound("template ${body.templateId} not found")
```

Reply:
I think that's harder to understand tbh 😅